java/src/com/ibm/streamsx/rest/Operator.java

```diff
@@ -50,13 +51,27 @@
     @Expose
     private String self;
 
-    /**
-     * this function is not intended for external consumption
-     */
-    void setConnection(final StreamsConnection sc) {
+    private void setConnection(final StreamsConnection sc) {
         connection = sc;
     }
 
+    static final List<Operator> getOperatorList(StreamsConnection sc, String operatorsList) {
+        List<Operator> opList;
+        OperatorArray opArray;
+        try {
+            opArray = new GsonBuilder().excludeFieldsWithoutExposeAnnotation().create().fromJson(operatorsList,
+                    OperatorArray.class);
+
+            opList = opArray.operators;
+            for (Operator op : opList) {
+                op.setConnection(sc);
+            }
+        } catch (IllegalStateException e) {
+            opList = new ArrayList<Operator>();
```

Comment:
hmm... my view has always been to fail gracefully where possible.  These are all the list retrieval operations that do this, not the individual ones.  In these cases, most users are likely to just iterate over the list, and move to the next level.  If the gson fails to parse, we won't trigger a full failure by throwing a runtime exception that users are not likely to have setup a catch for.   So for the common use case of just retrieving metrics of the various components and iterating through the list of objects, any object change will trigger a partial failure until a recompile is done.